openedx/core/djangoapps/profile_images/tests/test_images.py

```diff
@@ -146,24 +147,54 @@ def test_generation(self, image_type, dimensions):
             100: "hundred.jpg",
             1000: "thousand.jpg",
         }
-        mock_storage = mock.Mock()
         with make_uploaded_file(dimensions=dimensions, extension=extension, content_type=content_type) as uploaded_file:
-            with mock.patch(
-                "openedx.core.djangoapps.profile_images.images.get_profile_image_storage",
-                return_value=mock_storage,
-            ):
-                create_profile_images(uploaded_file, requested_sizes)
-                names_and_files = [v[0] for v in mock_storage.save.call_args_list]
-                actual_sizes = {}
-                for name, file_ in names_and_files:
-                    # get the size of the image file and ensure it's square jpeg
-                    with closing(Image.open(file_)) as image_obj:
-                        width, height = image_obj.size
-                        self.assertEqual(width, height)
-                        self.assertEqual(image_obj.format, 'JPEG')
-                        actual_sizes[width] = name
-                self.assertEqual(requested_sizes, actual_sizes)
-                mock_storage.save.reset_mock()
+            names_and_files = self._create_mocked_profile_images(uploaded_file, requested_sizes)
+            actual_sizes = {}
+            self.assertEqual(len(names_and_files), 3)
+            for name, file_ in names_and_files:
+                # get the size of the image file and ensure it's square jpeg
+                with closing(Image.open(file_)) as image_obj:
+                    width, height = image_obj.size
+                    self.assertEqual(width, height)
+                    self.assertEqual(image_obj.format, 'JPEG')
+                    actual_sizes[width] = name
+            self.assertEqual(requested_sizes, actual_sizes)
+
+    def test_jpeg_with_exif_orientation(self):
+        requested_images = {10: "ten.jpg"}
+        rotate_90_clockwise = 8  # Value used in EXIF Orientation field.
+        with make_image_file(orientation=rotate_90_clockwise, extension='.jpg') as imfile:
+            names_and_files = self._create_mocked_profile_images(imfile, requested_images)
+        self.assertEqual(len(names_and_files), 1)
+        for file_ in [nf[1] for nf in names_and_files]:
+            with closing(Image.open(file_)) as image_obj:
+                self.assertEqual(image_obj.format, 'JPEG')
+                self.assertIn('exif', image_obj.info)
+                self.assertEqual(_get_exif_orientation(image_obj.info['exif']), rotate_90_clockwise)
+
+    def test_jpeg_without_exif_orientation(self):
```

Comment:
Any thoughts on reducing code duplication between this test method and the one above?